lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -291,6 +291,25 @@ ::mlir::LogicalResult mlir::tt::ttir::MatmulOp::verify() {
 }
 // ANCHOR_END: adding_an_op_matmul_ttir_verify
 
+::mlir::LogicalResult mlir::tt::ttir::Conv2dOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType weightType = getWeight().getType();
+
+  if (inputType.getRank() < 3) {
+    return emitOpError("Input must be at least a 3D tensor");
+  }
+  if (weightType.getRank() != 4) {
+    return emitOpError("Weight must be a 4D tensor");
+  }
+  if (!getODSOperands(2).empty()) {
+    ::mlir::RankedTensorType biasType = getBias().getType();
```

Comment:
There should be a `getBias()` method or similar in `TTIROps.h.inc`. Here's an example for an optional dtype attr I have locally for an op I added.
```
  ::mlir::tt::DataTypeAttr getDtypeAttr() {
    auto attr = ::llvm::dyn_cast_or_null<::mlir::tt::DataTypeAttr>(getProperties().dtype);
    return attr;
  }
```

Can you check if that'd work here?

Reply:
It did